File: weather-library/src/main/java/org/rcgonzalezf/weather/openweather/model/ForecastData.java
package org.rcgonzalezf.weather.openweather.model;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;

public class ForecastData {

  @SerializedName("city")
  private City mCity;
  @SerializedName("cnt")
  private int mCount;
  @SerializedName("list")
  private List<WeatherData> mWeatherList;


  public ForecastData(
      City city,
      int count) {
    setCity(city);
    setCount(count);
    mWeatherList = new ArrayList<>(count);
  }

  @Override public String toString() {
    return "ForecastData [name=" + getCity().getName() + ", count=" + getCount() +"";
  }

  public City getCity() {
    return mCity;
  }

  public void setCity(City mCity) {
    this.mCity = mCity;
  }

  public int getCount() {
    return mCount;
  }

  public void setCount(int count) {
    mCount = count;
  }

  public List<WeatherData> getWeatherList() {
    return mWeatherList;
  }

  public void addWeatherItem(WeatherData weatherData) {
    mWeatherList.add(weatherData);
  }

}

File: weather-library/src/main/java/org/rcgonzalezf/weather/openweather/model/ForecastData.kt
package org.rcgonzalezf.weather.openweather.model

import com.google.gson.annotations.SerializedName
import org.rcgonzalezf.weather.common.models.converter.Data
import java.util.ArrayList

open // open for Mockito
class ForecastData(@SerializedName("city") var city: City, count: Int) : Data {

    @SerializedName("cnt")
    var count = 0

    @SerializedName("list")
    private val weatherList: MutableList<WeatherData>
    override fun toString(): String {
        return "ForecastData [name=" + city.name + ", count=" + count + ""
    }

    fun getWeatherList(): List<WeatherData> {
        return weatherList
    }

    fun addWeatherItem(weatherData: WeatherData) {
        weatherList.add(weatherData)
    }

    init {
        this.count = count
        weatherList = ArrayList(count)
    }
}
